### PR TITLE
Fix koa ValidateError

### PR DESCRIPTION
**Pull request by hpx7**
Closes https://github.com/lukeautry/tsoa/issues/1333

### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [X] This PR is associated with an existing issue?

**Closing issues**

Put `closes #XXXX` (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

I have verified this patch in my local tsoa + koa project

### CODE REVIEW

**Comment by hpx7** [comment]
@WoH would you be able to provide any guidance on the testing strategy for this?

**Comment by hpx7** [comment]
The `Koa Server (with noImplicitAdditionalProperties turned on)` tests are failing on master for me, it doesn't seem related to this PR

**Comment by WoH** [comment]
> The `Koa Server (with noImplicitAdditionalProperties turned on)` tests are failing on master for me, it doesn't seem related to this PR

I think most of the invalid JSON errors are a consequence tbh.

File: packages/cli/src/routeGeneration/templates/koa.hbs [hunk 1]
```diff
@@ -92,7 +92,7 @@ export function RegisterRoutes(router: KoaRouter) {
             } catch (err) {
               const error = err as any;
               context.status = error.status;
-              context.throw(error.status, JSON.stringify({ fields: error.fields }));
+              context.throw(context.status, error.message, error);
```
Comment (WoH) [inline packages/cli/src/routeGeneration/templates/koa.hbs:95]: I think we can just rethrow, right?
And set expose to true:
https://github.com/koajs/koa/blob/master/docs/api/context.md#ctxthrowstatus-msg-properties

**Comment by hpx7** [comment]
The current test failures seem related to some dependency upgrade. `yarn install && yarn test` works but `yarn install --no-lockfile && yarn test` fails. Curiously, `yarn list` seems to give back the same output either way.

Any ideas @WoH?

**Comment by WoH** [comment]
Yup, seems unrelated. I'll check what went wrong.

**Comment by hpx7** [comment]
Any luck so far @WoH ? Let me know if I can help

**Pull request merged**